web/src/pm_list_data.ts

```diff
@@ -22,12 +22,18 @@ export function get_active_user_ids_string(): string | undefined {
         return undefined;
     }
 
-    const emails = filter.operands("dm")[0];
+    let emails = filter.operands("dm")[0];
 
     if (!emails) {
         return undefined;
     }
+    const emails_list = emails.split(",");
+    const email_index = emails_list.indexOf(people.my_current_email());
 
+    if (email_index !== -1) {
+        emails_list.splice(email_index, 1);
+    }
+    emails = emails_list.join(",");
     return people.emails_strings_to_user_ids_string(emails);
```

Comment:
This function is also used here:

```
    if (narrow_to_private_messages_section) {                                                           
        const current_user_ids_string = pm_list_data.get_active_user_ids_string();                      
        if (current_user_ids_string !== undefined) {                                                    
            const $active_filter_li = $(                                                                
                `li[data-user-ids-string='${CSS.escape(current_user_ids_string)}']`,                    
            );                                                                                          
            scroll_pm_into_view($active_filter_li);                                                     
        }                                                                                               
        update_private_messages();                                                                      
```

How is `data-user-ids-string` set? Does stripping the current user's ID correct a bug there, too?

Reply:
I think the answer is that they're all generated with `pm_with_user_ids`, which does stripe the current user.

Reply:
OK but I think that suggests a cleaner implementation that calls `sorted_other_user_ids` on the result of `emails_strings_to_user_ids_string`, rather than doing all this email-address parsing here.

Want to try that and confirm it works correctly?

Reply:
`get_active_user_ids_string` is the right place as it will try to scroll to position when narrowing on a pm with `current-user-ids-string` including the current user id too. But since there is actually no pm which has current user in `data-user-ids-string` so the `scroll_pm_into_view` won't work.